gem5-configs/configs-gapbs-tests/gapbs_config.py

```diff
@@ -0,0 +1,122 @@
+
+""" Script to run GAP Benchmark suites workloads. 
+    The workloads have two modes: synthetic and real graphs.
+"""
+
+import sys
+import time
+
+import m5
+import m5.ticks
+from m5.objects import *
+
+sys.path.append('gem5/configs/common/') # For the next line...
+import SimpleOpts
+
+from system import *
+
+SimpleOpts.set_usage(
+    "usage: %prog [options] kernel disk cpu_type num_cpus mem_sys benchmark synthetic graph")
+
+SimpleOpts.add_option("--allow_listeners", default=False, action="store_true",
+                      help="Listeners disabled by default")
+
+
+def writeBenchScript(dir, benchmark_name, size, synthetic):
+    """
+    This method creates a script in dir which will be eventually
+    passed to the simulated system (to run a specific benchmark
+    at bootup).
+    """
+    input_file_name = '{}/run_{}_{}'.format(dir, benchmark_name, size)
+    if (synthetic):
+        with open(input_file_name,"w") as f:
+           f.write('./{} -g {}\n'.format(benchmark_name, size))
+    elif(synthetic==0):
+        with open(input_file_name,"w") as f:
+           f.write('#!/bin/sh\n')
+           f.write('./{} -sf {}'.format(benchmark_name, size))
+    
+    return input_file_name
+
+if __name__ == "__m5_main__":
+    (opts, args) = SimpleOpts.parse_args()
+
+    # create the system we are going to simulate
+    if len(args) != 8:
+        SimpleOpts.print_help()
+        m5.fatal("Bad arguments")
+    
+    kernel, disk, cpu_type, num_cpus, mem_sys, benchmark, synthetic, benchmark_size = args
+    num_cpus = int(num_cpus)
+
+    if (mem_sys == "classic"):
+        system = MySystem(kernel, disk, cpu_type, num_cpus,opts)
+    elif (mem_sys == "MI_example"):
+        system = MyRubySystem(kernel, disk, cpu_type, mem_sys, num_cpus,opts)
+
+    
+    system = MySystem(kernel, disk, cpu_type, num_cpus,opts)
+    benchmark_name = benchmark
+
+    output_dir = os.path.join(m5.options.outdir, "speclogs")
+
+    # For workitems to work correctly
+    # This will cause the simulator to exit simulation when the first work
+    # item is reached and when the first work item is finished.
+    system.work_begin_exit_count = 1
+    system.work_end_exit_count = 1
+
+    # Read in the script file passed in via an option.
+    # This file gets read and executed by the simulated system after boot.
+    # Note: The disk image needs to be configured to do this.
+
+    system.readfile = writeBenchScript(m5.options.outdir, benchmark_name,
+                                       benchmark_size, synthetic) 
+
+    # set up the root SimObject and start the simulation
+    root = Root(full_system = True, system = system)
+
+    if system.getHostParallel():
+        # Required for running kvm on multiple host cores.
+        # Uses gem5's parallel event queue feature
+        # Note: The simulator is quite picky about this number!
+        root.sim_quantum = int(1e9) # 1 ms
+
+    # instantiate all of the objects we've created above
+        m5.instantiate()
```

Comment:
This whitespace isn't correct... I guess this was only tests with kvm? ;)

Reply:
I only tested it with kvm. I will add more cpu models by tomorrow